packages/cli/src/config/settings.ts

```diff
@@ -111,18 +119,21 @@ export class LoadedSettings {
     user: SettingsFile,
     workspace: SettingsFile,
     errors: SettingsError[],
+    isTrusted?: boolean,
   ) {
     this.system = system;
     this.user = user;
     this.workspace = workspace;
     this.errors = errors;
+    this.isTrusted = isTrusted;
     this._merged = this.computeMergedSettings();
   }
 
   readonly system: SettingsFile;
   readonly user: SettingsFile;
   readonly workspace: SettingsFile;
   readonly errors: SettingsError[];
+  readonly isTrusted: boolean | undefined;
```

Comment:
nit: make this bool instead of `boolean | undefined`
only allow bools to be undefined when strictly required. this seems like a true boolean.